### PR TITLE
Added Wait-SeDriver -Script

### PR DESCRIPTION
**Pull request by ili101**
I don't know if you will like to add this  but I find it useful
I was looking on how to wait on class and saw that it is not available in the .NET Selenium version and you need to make you own wait condition or use XPath.
I tried the custom Wait and it turned out very nice so I added it to `Wait-SeDriver` so you can now wait on a scriptblock like this:
``` powershell
Wait-SeDriver -Script { (Get-SeElement -By ClassName 'active').Text -eq 'All Customers' }
```
I didn't run the tests (only tested it on a copy of the module and then copied the code to the git clone that you already updated by that time) so hopefully it's not broken

### CODE REVIEW

**Comment by itfranck** [comment]
@ili101 I am not sure I see the benefit and I will tell you why. 

Get-SeElement with the `-Timeout` parameter already do pretty much what using the `-Script` parameter of `Wait-SeDriver` do but with an extra benefit. It actually return the resource it found. If any. 

```
Get-SeElement -By ClassName -Value 'homeitem' -Timeout 3 -ErrorAction Continue
```

There's also the new (as new and undocumented as Wait-SeDriver) `Wait-SeElement` that can validate if an element exist

```
Wait-SeElement  -By XPath 'Something' -Condition ElementExists  -Timeout 10
```


So I am not convinced that Wait-SeDriver actually bring anything on top of that. 

```
Wait-SeDriver -Script { (Get-SeElement -By ClassName 'homeitem').Text -like '*Great*' }
```

I noticed you are evaluating a condition within the scriptblock so it is kind of a shorthand (one liner) to something that would otherwise look like : 

```
$MyItems = Get-SeElement -By ClassName -Value 'homeitem' -Timeout 3 -ErrorAction Continue
$GreatItems = $MyItems | Where Text -Like '*Great*' 
if ($GreatItems -ne $null) {
    # Do something here
}
```

which is great, but since you don't get the relevant resources, you'd end up with at least a second Get-SeElement call to get the actual resource if you need to do anything else than knowing about whether it exists or not. 

On a related note, I was looking at expanded capabilities through #116  to support something that would allow you to get a more refined result through Get-SeElement so you could use it like this : 

```
$MyItems = Get-SeElement -By ClassName -Value 'homeitem' -Filter {$_.Text -like '*Great*'} -Timeout 5 
if ($null -ne $MyItems) {
    #Do something
}
```

Combined with the existing #133 attributes parameter, you could even go futher and do something like : 
```
$FilterBlock = { $_.Attributes.Name -like '*Great*' -or $_.Attributes.Id -eq 'SearchF' }
$MyItems = Get-SeElement -By ClassName -Value 'homeitem' -Filter $FilterBlock -Attributes @(name, id) -Timeout 5 
if ($null -ne $MyItems) {
    #Do something
}
```

tl;dr

I think `Get-SeElement -Timeout` and `Wait-SeElement` pretty much already cover what this Pull request aim to do without adding real benefits. If you think I am missing something though, please let me know. 

I am curious to hear about your thoughts.

P.s: you can view the status tests here: 
https://dev.azure.com/adamrdriscoll/Selenium/_build?definitionId=25
Tests are performed against all pull requests. 
Note that some are failing intermittently no matter what currently (Mostly the failing tests come from Frame switching / alert dismissing ) 

P.P.S : 

This PR made me find a bug with Get-SeElement and Timeout as it looks like only the first element is returned when used with timeout. I'll open an issue for that.

**Comment by ili101** [comment]
Not exactly. In this particular situation I need to know that the page was updated and I can continue, I don't actually need the element, and the element need to have the text and the class not just one of them.
But as you said I can work around it by doing something like:
``` powershell
    $TimeOut = 5
    $StartTime = Get-Date
    While ((Get-SeElement -By ClassName 'active').Text -ne 'All Customers') {
        Start-Sleep -Seconds 1
        if ($StartTime.AddSeconds($TimeOut) -lt (Get-Date)) {
            Throw 'Timeout'
        }
    }
```
Or just do the same thing manually, it's just a little bit longer:
``` powershell
    $TimeOut = 5
    $WebDriverWait = [OpenQA.Selenium.Support.UI.WebDriverWait]::new($Driver, (New-TimeSpan -Seconds $Timeout))
    $WebDriverWait.Until(
        [System.Func[OpenQA.Selenium.IWebDriver, Bool]] {
            (Get-SeElement -By ClassName 'active').Text -eq 'All Customers'
        }
    )
```
Anyway this is probably an edge case that is not that common and can be workaround fairly easily.

**Comment by itfranck** [comment]
@ili101 

I see. 
No, that kind of make sense after all. 

I had in mind something very similar for Get-SeElement with a Filter parameter, which in combination with the existing Timeout, would produce the same result as the suggested PR (but return the elements) 

 `[Void](Get-SeElement -By ClassName 'active' -Filter {$_.Text -eq 'All Customers'} -Timeout 10)`

That being said, Get-SeElement and Wait-SeDriver /  Wait-SeElement overlap in a few place already.

I might look into reformatting it a tiny bit to 

add **ScriptBlock** as a condition (1)  instead of having an additional parameter set so it get the friendly tooltip readily available. 

image

**Comment by ili101** [comment]
Yes, adding it to the existing parameters makes more sense.
Regarding the output. It can actually output if you need. Just replace  the `Bool` with `Object` here `[System.Func[OpenQA.Selenium.IWebDriver, Object]]` and return somting from the script block.
I used bool as `Wait-SeDriver` already discard the output and replace it with bool, but you can change that or use that method to make `Get-SeElement -Filter -Timeout 10` work, although it can be confusing if the user want to wait on the element then filter it or wait on the element and the filter. maybe separate this to 2 modes somehow?
